[Bug]: Crashlooping topic-operator due to invalid logging configuration

### Bug Description

The topic-operator in our clusters is crashing due to invalid logging configuration.

The exception thrown by the topic-operator:
```
Caused by: org.apache.logging.log4j.core.config.ConfigurationException: No name attribute provided for Logger clients
    at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.createLogger(PropertiesConfigurationBuild
    at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.build(PropertiesConfigurationBuilder.java
    at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory.getConfiguration(PropertiesConfigurationF
    at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory.getConfiguration(PropertiesConfigurationF
    at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:525)
    at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:500)
    at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:424)
    at org.apache.logging.log4j.core.config.ConfigurationFactory.getConfiguration(ConfigurationFactory.java:325)
    at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:690)
    at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:711)
    at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:253)
    at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:155)
    at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:47)
    at org.apache.logging.log4j.LogManager.getContext(LogManager.java:196)
    at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:599)
    at io.strimzi.operator.common.ReconciliationLogger.create(ReconciliationLogger.java:66)
    at io.strimzi.operator.topic.TopicOperatorMain.<clinit>(TopicOperatorMain.java:41)
```

The topic operator config map data for the topic-operator that fails is as follows:
```yaml
data:
  log4j2.properties: |
    # Do not change this generated file. Logging can be configured in the corresponding Kubernetes resource.
    name=TOConfig
    appender.console.type=Console
    appender.console.name=STDOUT
    appender.console.layout.type=PatternLayout
    appender.console.layout.pattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
    rootLogger.level=INFO
    rootLogger.appenderRefs=stdout
    rootLogger.appenderRef.console.ref=STDOUT
    rootLogger.additivity=false
    logger.jetty.name=org.eclipse.jetty
    logger.jetty.level=INFO
    logger.jetty.additivity=false
    logger.clients.level=INFO
    logger.toc.level=INFO
    logger.toc.name=io.strimzi.operator.topic.TopicOperator
    logger.top.level=INFO
    logger.top.name=io.strimzi.operator.topic

    monitorInterval=30
```

We also have a cluster for which the topic-operator does work and the only difference is `logger.clients.level=INFO`. The cluster that works does not have this line. The clusters that crash do have this line.

### Steps to reproduce

I don't know what triggers this exception

### Expected behavior

That the topic-operator starts.

### Strimzi version

0.45.0

### Kubernetes version

1.30.5

### Installation method

Helm chart

### Infrastructure

Self-hosted Kubernetes (on OpenStack)

### Configuration files and logs

Operator HelmRelease YAML
```yaml
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: strimzi-kafka-release
  namespace: edusrc
spec:
  install:
    crds: CreateReplace
    remediation:
      retries: 10
  upgrade:
    crds: CreateReplace
    remediation:
      retries: 10
      remediateLastFailure: true
  uninstall:
    keepHistory: true
  releaseName: strimzi-kafka
  targetNamespace: infra-strimzi
  storageNamespace: infra-strimzi
  interval: 30m
  kubeConfig:
    secretRef:
      name: ${kubeconfigSecret}
  driftDetection:
    mode: enabled
  chart:
    spec:
      chart: strimzi-kafka-operator
      version: "0.45.0"
      sourceRef:
        kind: HelmRepository
        name: strimzi-helmrepo
      interval: 12h
  values:
    extraEnvs:
    - name: STRIMZI_POD_SECURITY_PROVIDER_CLASS
      value: restricted
    podSecurityContext:
      fsGroup: 1000
      runAsGroup: 1000
      runAsUser: 1000
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      readOnlyRootFilesystem: true
      runAsNonRoot: true
      seccompProfile:
        type: RuntimeDefault
      runAsUser: 1000
    watchNamespaces: [ 'infra-kafka' ]
    replicas: 3
    dashboards:
      enabled: false
```

Cluster HelmRelease YAML
```yaml
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaNodePool
metadata:
  name: shared-cluster-controller-nodes
  namespace: infra-kafka
  labels:
    strimzi.io/cluster: shared-cluster
spec:
  replicas: 3 # https://docs.confluent.io/platform/current/installation/system-requirements.html#cp-hardware
  roles:
    - controller
  storage:
    type: jbod
    volumes:
      - id: 0
        type: persistent-claim
        size: 10Gi
        kraftMetadata: shared
        deleteClaim: false
        class: hdd-retain
  resources:
    requests:
      memory: 512Mi
      cpu: "0.5"
    limits:
      memory: 512Mi
      cpu: "0.5"
---
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaNodePool
metadata:
  name: shared-cluster-broker-nodes
  namespace: infra-kafka
  labels:
    strimzi.io/cluster: shared-cluster
spec:
  replicas: 3
  roles:
    - broker
  storage:
    type: jbod
    volumes:
      - id: 0
        type: persistent-claim
        size: 30Gi
        deleteClaim: false
        class: hdd-retain
  resources:
    requests:
      memory: 1024Mi
      cpu: "0.5"
    limits:
      memory: 1024Mi
      cpu: "0.5"
---
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: shared-cluster
  namespace: infra-kafka
  annotations:
    strimzi.io/node-pools: enabled
    strimzi.io/kraft: enabled
spec:
  kafka:
    version: 3.8.0
    metadataVersion: 3.8-IV0
    listeners:
      - name: tls
        port: 9093
        type: internal
        tls: true
        authentication:
          type: tls
    authorization:
      type: simple
      superUsers:
        - CN=kafka-super-user
    readinessProbe:
      initialDelaySeconds: 15
      timeoutSeconds: 5
    livenessProbe:
      initialDelaySeconds: 15
      timeoutSeconds: 5
    config:
      auto.create.topics.enable: "false"
      delete.topic.enable: "false" # used as safety mechanism, push a "true" value to enable deletion.
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      default.replication.factor: 3
      min.insync.replicas: 2
      log.cleanup.policy: "compact"
      log.retention.ms: -1 # never expire based on time
      log.retention.bytes: -1  # never expire based on size
      log.cleaner.delete.retention.ms: 86400000 # only process tombstones after a day
    logging:
      type: inline
      loggers:
        kafka.root.logger.level: INFO
        log4j.logger.kafka.coordinator.transaction: TRACE
        log4j.logger.kafka.log.LogCleanerManager: INFO
        log4j.logger.kafka.request.logger: INFO
        log4j.logger.io.strimzi.kafka.oauth: INFO
        log4j.logger.org.openpolicyagents.kafka.OpaAuthorizer: INFO
    metricsConfig:
      type: jmxPrometheusExporter
      valueFrom:
        configMapKeyRef:
          name: kafka-metrics-config
          key: kafka-metrics-config.yml
  entityOperator:
    topicOperator:
      resources:
        requests:
          memory: 256Mi
          cpu: "250m"
        limits:
          memory: 256Mi
          cpu: "250m"
      watchedNamespace: infra-kafka
      reconciliationIntervalMs: 60000
      logging:
        type: inline
        loggers:
          rootLogger.level: INFO
          logger.top.name: io.strimzi.operator.topic
          logger.top.level: INFO
          logger.toc.name: io.strimzi.operator.topic.TopicOperator
          logger.toc.level: INFO
          logger.clients.level: INFO
    userOperator:
      resources:
        requests:
          memory: 256Mi
          cpu: "250m"
        limits:
          memory: 256Mi
          cpu: "250m"
      logging:
        type: inline
        loggers:
          rootLogger.level: INFO
          logger.uop.name: io.strimzi.operator.user
          logger.uop.level: INFO
          logger.abstractcache.name: io.strimzi.operator.user.operator.cache.AbstractCache
          logger.abstractcache.level: TRACE
          logger.jetty.level: INFO
  kafkaExporter:
    topicRegex: ".*"
    groupRegex: ".*"
    logging: info
```

### Additional context

_No response_

--- a/documentation/api/io.strimzi.api.kafka.model.kafka.entityoperator.EntityTopicOperatorSpec.adoc
+++ b/documentation/api/io.strimzi.api.kafka.model.kafka.entityoperator.EntityTopicOperatorSpec.adoc
@@ -42,15 +42,12 @@ spec:
           logger.top.level: DEBUG # <2> 
           logger.toc.name: io.strimzi.operator.topic.TopicOperator # <3>
           logger.toc.level: TRACE # <4>
-          logger.clients.level: DEBUG # <5>
   # ...
 ----
 <1> Creates a logger for the `topic` package.
 <2> Sets the logging level for the `topic` package.
 <3> Creates a logger for the `TopicOperator` class.
 <4> Sets the logging level for the `TopicOperator` class.
-<5> Changes the logging level for the default `clients` logger. The `clients` logger is part of the logging configuration provided with Strimzi.
-By default, it is set to `INFO`.   
 
 NOTE: When investigating an issue with the operator, it's usually sufficient to change the `rootLogger` to `DEBUG` to get more detailed logs. 
 However, keep in mind that setting the log level to `DEBUG` may result in a large amount of log output and may have performance implications.

--- a/mockkube/src/test/resources/log4j2.properties
+++ b/mockkube/src/test/resources/log4j2.properties
@@ -1,4 +1,4 @@
-name = STConfig
+name = MockKubeConfig
 
 appender.console.type = Console
 appender.console.name = STDOUT
@@ -9,7 +9,3 @@ rootLogger.level = ${env:STRIMZI_LOG_LEVEL:-INFO}
 rootLogger.appenderRefs = stdout
 rootLogger.appenderRef.console.ref = STDOUT
 rootLogger.additivity = false
-
-logger.clients.name = org.apache.kafka.clients
-logger.clients.level = info
-

--- a/operator-common/src/test/resources/log4j2.properties
+++ b/operator-common/src/test/resources/log4j2.properties
@@ -1,4 +1,4 @@
-name = STConfig
+name = OperatorCommonConfig
 
 appender.console.type = Console
 appender.console.name = STDOUT
@@ -9,7 +9,3 @@ rootLogger.level = ${env:STRIMZI_LOG_LEVEL:-INFO}
 rootLogger.appenderRefs = stdout
 rootLogger.appenderRef.console.ref = STDOUT
 rootLogger.additivity = false
-
-logger.clients.name = org.apache.kafka.clients
-logger.clients.level = info
-

--- a/systemtest/src/main/resources/log4j2.properties
+++ b/systemtest/src/main/resources/log4j2.properties
@@ -25,7 +25,7 @@ rootLogger.appenderRef.rolling.level = DEBUG
 rootLogger.additivity = false
 
 logger.clients.name = org.apache.kafka.clients
-logger.clients.level = info
+logger.clients.level = INFO
 
 logger.fabric8.name = io.fabric8.kubernetes.client
 logger.fabric8.level = OFF

